src/functions/fetchRequests/fetchRequestsUser.js

```diff
@@ -0,0 +1,257 @@
+
+// createUser (userObj, setUser)        :: createUser( {userObj}, setUser ) => fetch(); use setUser( newUser ); save token
+// updateUser (userObj, setUser)        :: updateUser( {userObj}, setUser, token ) => fetch(); use setUser( user ); 
+// deleteUser (userObj, setUser)        :: deleteUser( {userObj,password}, token ) => fetch(); setUser( {} ); remove token;
+
+// retrieveUser (setUser)               :: retrieveUser( token ) => fetch(token); retrieve userObj
+// retrieveUserByUsername ( username, setUser) :: retrieveUserByUsername( username ) => fetch(token); retrieve userObj
+// retrieveUserFriends (setFriendList)  :: retrieveUserFriends( setFriends ) => fetch(token); retrieve [userObj,]
+
+// ------------------------------------------------------
+
+
+// createUser( {userObj}, setUser ) => fetch(); use setUser( newUser ); save token
+export const createUser = async (userObj, setUser) => {
+
+    try {
+        // POST /user {userObj} => {userObj, token}
+        const res = await fetch(`${process.env.REACT_APP_REST_API}/user`, {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify( userObj ),
+        });
+
+
+        const data = await res.json();
+        console.log('-> createUser(), data:', data);
+
+        if (!res.ok || data.error) throw new Error(`FetchError ${res.status} : ${data.error || data}`)
+
+        // since Michael doesn't want the token in localStorage, attach it to userObj
+        data.user.authToken = data.token
+
+        // set the user to state
+        setUser( data.user );
+
+        // return the result
+        return res.ok;
+
+    } catch (error) {
+        console.log('-> createUser(), error: ',error);
+        return false;
+    }
+
+};
+
+
+// updateUser( {userObj}, setUser, token ) => fetch(); use setUser( user ); 
+export const updateUser = async (userObj, setUser) => {
+    try {
+        // PATCH /user {userObj, token} => {userObj}
+        const res = await fetch(`${process.env.REACT_APP_REST_API}/user`, {
+            method: "PATCH",
+            headers: { 
+                "Content-Type": "application/json",
+                "authorization": userObj?.authToken,
+            },
+            body: JSON.stringify( userObj ),
+        });
+
+        const data = await res.json();
+        console.log('-> updateUser(), data:', data);
+
+        // if (data.error) throw new Error(data.error)
+        if (!res.ok || data.error) throw new Error(`FetchError ${res.status} : ${data.error || data}`)
+
+        // set the user to state -- not needed i think
+        setUser( data );
+
+        // returns the result
+        return res.ok;
+
+    } catch (error) {
+        console.log('-> updateUser(), error: ',error);
+        return false;
+    }
+};
+
+
+// deleteUser( {userObj,password}, token ) => fetch(); setUser( {} ); remove token;
+export const deleteUser = async (userObj, setUser) => {
```

Comment:
We don't yet know how the user will be deleted, if at all. Please remove for now.